### PR TITLE
Fix Checkbox with error state

### PR DESCRIPTION
**Pull request by RoyEJohnson**
[CORE-77]
I activated the React eslint plugins and made adjustments so that everything is happy. We have 100% code coverage.

[CORE-77]: https://openstax.atlassian.net/browse/CORE-77?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/forms/uncontrolled/inputTypes.tsx [hunk 1]
```diff
@@ -232,7 +231,7 @@ export const Checkbox = ({
     </CheckboxLine>
     <HelpText value={help} />
     {error !== undefined && error.length > 0 && 
-      <StyledErrorMessage>This activity has been deleted. Please deselect this activity to remove it from this assignment and avoid errors.</StyledErrorMessage>
+      <StyledErrorMessage>{error[0]}</StyledErrorMessage>
```
Comment (bethshook) [inline src/components/forms/uncontrolled/inputTypes.tsx:234]: I think we might want to map the messages

File: src/components/Error.tsx [hunk 2]
```diff
@@ -39,7 +39,7 @@ export const Error = ({ heading, children, ...props }: ErrorPropTypes) => {
   return <ModalBody {...props} data-testid='error'>
     <ModalBodyHeading>{heading ?? `Uh-oh, there's been a glitch`}</ModalBodyHeading>
     {children ?? <>
-      We're not quite sure what went wrong. Restart your browser. If this doesn't solve
+      We&apos;re not quite sure what went wrong. Restart your browser. If this doesn&apos;t solve
```
Comment (bethshook) [inline src/components/Error.tsx:42]: is there a config we can change so we don't have to do this?

Comment (RoyEJohnson) [reply src/components/Error.tsx:42]: This is for the ["no-unescaped-entities" rule](https://github.com/jsx-eslint/eslint-plugin-react/blob/master/docs/rules/no-unescaped-entities.md). It is turn-off-able. Its purpose is to catch where markup intended to be JSX gets included in text.
As an alternative to the HTML entity, I think we could use `{'}` for the apostrophe.

Comment (bethshook) [reply src/components/Error.tsx:42]: hmm how do we handle the rule on the other projects?

Comment (RoyEJohnson) [reply src/components/Error.tsx]: It doesn't look like we have the react plugins for eslint in rex or assignments.
I turned off the no-unescaped-entities rule.

File: src/components/forms/uncontrolled/inputTypes.tsx [hunk 3]
```diff
@@ -193,12 +193,11 @@ export const Radio = ({
...
  * checkbox element
  */
-type CheckboxProps = React.ComponentPropsWithoutRef<'input'> & InputProps & {
+type CheckboxProps = React.ComponentPropsWithoutRef<'input'> & InputProps & 
+Parameters<typeof StyledCheckbox>[0] & {
   onChangeValue?: (value: boolean | undefined) => void;
   wrapperProps?: React.ComponentPropsWithoutRef<'label'>;
   error?: string[];
-  size?: CheckboxSize;
-  variant?: CheckboxVariant;
```
Comment (bethshook) [inline src/components/forms/uncontrolled/inputTypes.tsx:201]: personally prefer it the way it was just so it's obvious what's in there

**Review by bethshook — APPROVED** [review]

**Comment by bethshook** [comment]
left a minor comment but feel free to merge

**Pull request merged**